outbound/hmail.js

```diff
@@ -205,7 +207,7 @@ class HMailItem extends events.EventEmitter {
                     // assume string
                     const matches = /^(.*?)(:(\d+))?$/.exec(mx);
                     if (!matches) {
-                        throw ("get_mx returned something that doesn't match hostname or hostname:port");
+                        throw (new Error("get_mx returned something that doesn't match hostname or hostname:port"));
```

Comment:
I agree that anything throwing an error should `throw new Error('....')`. But...we haven't always done that. Have you verified that any code that will catch this error is expecting an error object and not just a string?

Reply:
It passes the test suite, and I made this change aside from the other issues we were having with P42 if-else changes that I have Lars looking at.

Reply:
Our test suite isn't complete, and is not along good enough to justify a change like this. Either do the legwork or revert this change.

Reply:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw

If anything, we're not even utilizing the full power of Error here; there are newer ES20xx commands that can do error handling like you would in C#, Nim, or other languages. And I don't see much "catch" of throws, either. This still seems legit to me.

Reply:
You're preaching at the choir. The question was never whether or not this was legit or a good idea. The question is whether or not this BREAKS OUR CODE because the caller(s) of this function are expecting a different type of throw. If they are, more changes are required.

Reply:
@msimerson sorry I didn't see this sooner; I wasn't getting fully updated with notifications. I have everything else fixed so far; I'll dig into this in a bit. Thank you for your extreme patience.